OsmAnd/src/net/osmand/plus/myplaces/tracks/dialogs/FilterAllVariantsListFragment.kt

```diff
@@ -0,0 +1,212 @@
+package net.osmand.plus.myplaces.tracks.dialogs
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.Window
+import android.widget.ProgressBar
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.Toolbar
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.FragmentManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.appbar.AppBarLayout
+import net.osmand.plus.OsmandApplication
+import net.osmand.plus.R
+import net.osmand.plus.base.BaseOsmAndDialogFragment
+import net.osmand.plus.myplaces.tracks.DialogClosedListener
+import net.osmand.plus.myplaces.tracks.TrackFiltersHelper
+import net.osmand.plus.myplaces.tracks.filters.BaseTrackFilter
+import net.osmand.plus.myplaces.tracks.filters.SmartFolderUpdateListener
+import net.osmand.plus.myplaces.tracks.filters.TrackFilterPropertiesAdapter
+import net.osmand.plus.myplaces.tracks.filters.TracksCollectionFilter
+import net.osmand.plus.utils.AndroidUtils
+import net.osmand.plus.widgets.dialogbutton.DialogButton
+import studio.carbonylgroup.textfieldboxes.ExtendedEditText
+
+class FilterAllVariantsListFragment : BaseOsmAndDialogFragment(), SmartFolderUpdateListener {
+	companion object {
+		val TAG: String = FilterAllVariantsListFragment::class.java.simpleName
+
+		fun showInstance(
+			app: OsmandApplication,
+			manager: FragmentManager,
+			filter: BaseTrackFilter,
+			dialogClosedListener: DialogClosedListener?,
+			adapter: RecyclerView.Adapter<RecyclerView.ViewHolder>,
```

Comment:
You pass adapter from viewholder from previous screen to new fragment, looks strange